src/transformStateWithClones.js

```diff
@@ -6,8 +6,51 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  // const newObject = {};
+
+  const clone = {}; // новий порожній об'єкт
+  const resultArray = [];
+
+  for (const key in state) {
+    clone[key] = state[key];
+  }
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(clone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(clone, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clearProperties(clone);
+        break;
+    }
+    resultArray.push({ ...clone });
```

Comment:
You're correctly making a copy of the `clone` object before pushing it to the `resultArray`. However, the clone operation should be done once after the switch statement, not in every case, to avoid repetition.